pkg/mkimage-raw-efi/install

```diff
@@ -170,24 +171,82 @@ done
-   PDEV="/dev/$INSTALL_PERSIST"
-   dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc
-   sgdisk -Z --clear "$PDEV" 2>/dev/null || :
-   sgdisk --new 1:2048:0 --typecode=1:5f24425a-2dfa-11e8-a270-7b663faccc2c --change-name=1:P3 "$PDEV"
-   sgdisk -v "$PDEV"
-   # force make-raw to skip persist
-   MAKE_RAW_PARTS="efi imga imgb conf"
+   if echo "$INSTALL_PERSIST"| grep -q ","; then
+     modprobe zfs
+     MAKE_RAW_PARTS="efi imga imgb conf"
+     IFS=',' ;for dev in $INSTALL_PERSIST; do
+        if [ "$dev" = "$INSTALL_DEV" ]; then
+           # in case we want to have one of P3 on the bootable disk
+           MAKE_RAW_PARTS="efi imga imgb conf persist"
+           DISK_WITH_P3="$dev"
+           POOL_CREATION_COMMAND="$POOL_CREATION_COMMAND $P3_ON_BOOT_PLACEHOLDER"
+           DISKS_TO_MERGE_COUNT=$((DISKS_TO_MERGE_COUNT+1))
+        else
+           if [ -f "/sys/block/$dev/dev" ]; then
+              PDEV="/dev/$dev"
+              dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc && sync
+              sgdisk -Z --clear "$PDEV" 2>/dev/null || :
+              POOL_CREATION_COMMAND="$POOL_CREATION_COMMAND $PDEV"
+              DISKS_TO_MERGE_COUNT=$((DISKS_TO_MERGE_COUNT+1))
+           else
+              echo "WARNING: Cannot find /sys/block/$dev/dev, will skip it"
+           fi
+        fi
+     done
+   else
+     # apparently sgdisk -Z doesn't clear MBR and keeps complaining
+     PDEV="/dev/$INSTALL_PERSIST"
+     dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc
+     sgdisk -Z --clear "$PDEV" 2>/dev/null || :
+     sgdisk --new 1:2048:0 --typecode=1:5f24425a-2dfa-11e8-a270-7b663faccc2c --change-name=1:P3 "$PDEV"
+     sgdisk -v "$PDEV"
+     # force make-raw to skip persist
+     MAKE_RAW_PARTS="efi imga imgb conf"
+   fi
 fi
 
 # do the install (unless we're only here to collect the black box)
 grep -q eve_blackbox /proc/cmdline || /make-raw "/dev/$INSTALL_DEV" $MAKE_RAW_PARTS || bail "Installation failed. Entering shell..."
 
-# now the disk is ready - mount partitions
-mount_part P3 "$INSTALL_DEV" 2>/dev/null
+
+
+if [ "$POOL_CREATION_COMMAND" != "" ] && [ $DISKS_TO_MERGE_COUNT -gt 1 ]; then
+  if [ "$DISK_WITH_P3" != "" ]; then
+    P3_ID="$(find_part P3 "$DISK_WITH_P3")"
+    [ -z "$P3_ID" ] && bail "Installation failed. Cannot found P3. Entering shell..."
+    POOL_CREATION_COMMAND=$(echo "$POOL_CREATION_COMMAND" | sed "s#$P3_ON_BOOT_PLACEHOLDER#/dev/$P3_ID#g")
+    echo "WARNING: Will use /dev/$P3_ID instead of /dev/$DISK_WITH_P3 for P3 because of bootable parts needed"
+  fi
+  # we use mirror for 2 disks, raidz1 for 3 disks and raidz2 for more than 3 disks
+  case $DISKS_TO_MERGE_COUNT in
+    2)
+      POOL_CREATION_COMMAND="mirror $POOL_CREATION_COMMAND"
+      ;;
+    3)
+      POOL_CREATION_COMMAND="raidz1 $POOL_CREATION_COMMAND"
+      ;;
+    *)
+      POOL_CREATION_COMMAND="raidz2 $POOL_CREATION_COMMAND"
+      ;;
+  esac
+  [ -e /root/sys ] || mkdir /root/sys && mount -t sysfs sysfs /root/sys
```

Comment:
Ok, all of this needs to be either put into a function -- or better yet -- we need eve(1) exec-like script in the install container

Reply:
Will move it to a function as a first variant.

Reply:
Moved